homework/index.js

```diff
@@ -30,15 +33,189 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  const root = document.getElementById('root');
+  const container = createAndAppend('div', root, { id: 'container' });
+
+  function getSelectedData(value, data) {
+    let selected;
+    data.forEach(item => {
+      if (item.name === value) {
+        selected = item;
       }
     });
+    return selected;
+  }
+
+  function generateSelections(repos) {
+    const header = createAndAppend('header', root, {});
+    const selectionElem = createAndAppend('select', header, {
+      id: 'repositories',
+    });
+    let id = 0;
+
+    repos.forEach(repo => {
+      createAndAppend('option', selectionElem, {
+        id: id++,
+        value: repo,
+        text: repo,
+      });
+    });
+  }
+
+  function generateInfoSection(selected, data) {
+    const div = createAndAppend('div', container, { id: 'infoDiv' });
+    const selectedData = getSelectedData(selected, data);
+    createAndAppend('a', div, {
+      id: 'repoName',
+      text: `Repository: ${selectedData.name}`,
+      href: selectedData.html_url,
+      target: '_blank',
+    });
+    createAndAppend('p', div, {
+      id: 'desc',
+      text: `Description: ${selectedData.description}`,
+    });
+    createAndAppend('p', div, {
+      id: 'forks',
+      text: `Forks: ${selectedData.forks}`,
+    });
+    createAndAppend('p', div, {
+      id: 'updated_at',
+      text: `Updated: ${selectedData.updated_at}`,
+    });
+  }
+
+  function generateContributorsSection(selected) {
+    const contributorsDiv = createAndAppend('div', container, { id: 'contsDiv' });
+    createAndAppend('p', contributorsDiv, { id: 'cont-header', text: 'Contributors' });
+    const contributorsLists = createAndAppend('ul', contributorsDiv, { id: 'conts-list' });
+
+    fetchJSON(`https://api.github.com/repos/HackYourFuture/${selected}/contributors`).then(
+      data => {
+        data.forEach(item => {
+          const contributorListItem = createAndAppend('li', contributorsLists, {
+            id: `${item.login}ListItem`,
```

Comment:
The attributes in HTML elements should ideally be kept `kebab-case` unlike `camelCase` 🙂